### PR TITLE
Biggest commit you have ever seen

### PR DESCRIPTION
**Pull request by sahma19**

### CODE REVIEW

**Review by TopSwagCode — APPROVED** [review]

File: domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs [hunk 1]
```diff
@@ -159,20 +154,20 @@ public async Task SeedTransferAgreementProposals(IEnumerable<TransferAgreementPr
     public HttpClient CreateUnauthenticatedClient()
     {
         var client = CreateClient();
-        client.DefaultRequestHeaders.Add("EO_API_VERSION", "20231123");
+        client.DefaultRequestHeaders.Add("EO_API_VERSION", "20240103");
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs:157]: Would be nice with this static class with known versions, instead of magic strings all over the codebase.

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/Api/v2024_01_03/Controllers/TransferAgreementHistoryEntriesControllerTests.cs [hunk 2]
```diff
@@ -7,16 +7,16 @@
...
+using API.Transfer.Api.v2024_01_03.Dto.Responses;
 using FluentAssertions;
 using Newtonsoft.Json;
 using NSubstitute;
 using VerifyTests;
 using VerifyXunit;
 using Xunit;
 
-namespace API.IntegrationTests.Transfer.Api.v2023_11_23.Controllers;
+namespace API.IntegrationTests.Transfer.Api.v2024_01_03.Controllers;
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API.IntegrationTests/Transfer/Api/v2024_01_03/Controllers/TransferAgreementHistoryEntriesControllerTests.cs:19]: Same controller could support different versions, so version should be kept out of namespaces.

File: domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Controllers/TransferAgreementProposalController.cs [hunk 3]
```diff
@@ -19,26 +19,14 @@ namespace API.Transfer.Api.v2024_01_03.Controllers;
 [Authorize]
 [ApiController]
 [ApiVersion("20240103")]
-[Route("api/transfer-agreement-proposals")]
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Controllers/TransferAgreementProposalController.cs:22]: Has Alexander already migrated to front-end to the new endpoints?

File: domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Controllers/TransferAgreementProposalController.cs [hunk 4]
```diff
@@ -164,10 +152,6 @@ public async Task<ActionResult> DeleteTransferAgreementProposal(Guid id)
         }
 
         var user = new UserDescriptor(User);
-        if (proposal.ReceiverCompanyTin != null && user.Organization!.Tin != proposal.ReceiverCompanyTin)
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Controllers/TransferAgreementProposalController.cs:167]: I don't get this, what does this have to do with API versioning?

**Review by martinhenningjensen — CHANGES_REQUESTED** [review]

File: domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs (hunk 1 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs:157]: I was thinking about doing that in a separate commit. There is already alot happening in this commit.

File: domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Controllers/TransferAgreementProposalController.cs (hunk 4 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Controllers/TransferAgreementProposalController.cs:167]: Nothing.

File: domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs (hunk 1 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs:157]: Done:

https://github.com/Energinet-DataHub/energy-origin/pull/684/commits/761c9e64fb855ccc8e28e847b5c2986e0c35c998

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/Api/v2024_01_03/Controllers/TransferAgreementHistoryEntriesControllerTests.cs (hunk 2 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Transfer/Api/v2024_01_03/Controllers/TransferAgreementHistoryEntriesControllerTests.cs]: Done:

https://github.com/Energinet-DataHub/energy-origin/pull/684/commits/84de4b074372535c545f2314f9930f0aa7050a88

File: domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Controllers/TransferAgreementProposalController.cs (hunk 3 above)
Comment (martinhenningjensen) [reply domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Controllers/TransferAgreementProposalController.cs:22]: Won't this break the frontend then?

Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Controllers/TransferAgreementProposalController.cs:22]: Most likely.

**Review by martinhenningjensen — DISMISSED** [review]

**Review by frontend-specialisten — APPROVED** [review]

**Review by TopSwagCode — APPROVED** [review]
